src/analysis/buildTypeWeights.ts

```diff
@@ -19,168 +21,199 @@ import {
+    type: GraphQLObjectType | GraphQLInterfaceType,
+    typeWeightObject: TypeWeightObject,
+    typeWeights: TypeWeightConfig
+): Type {
+    const result: Type = {
+        fields: {},
+        // FIXME: assigning the weight will get busy when we add mutations/subscriptions. is there a better way ?
+        weight:
+            type.name === 'Query'
+                ? typeWeights.query || DEFAULT_QUERY_WEIGHT
+                : typeWeights.object || DEFAULT_OBJECT_WEIGHT,
+    };
+    const fields = type.getFields();
+
+    // Iterate through the fields and add the required data to the result
+    Object.keys(fields).forEach((field: string) => {
+        // The GraphQL type that this field represents
+        const fieldType: GraphQLOutputType = fields[field].type;
+        if (
+            isScalarType(fieldType) ||
+            (isNonNullType(fieldType) && isScalarType(fieldType.ofType))
+        ) {
+            result.fields[field] = {
+                weight: typeWeights.scalar || DEFAULT_SCALAR_WEIGHT,
+            };
+        } else if (
+            isInterfaceType(fieldType) ||
+            isUnionType(fieldType) ||
+            isEnumType(fieldType) ||
+            isObjectType(fieldType)
+        ) {
+            result.fields[field] = {
+                resolveTo: fieldType.name.toLocaleLowerCase(),
+            };
+        } else if (isListType(fieldType)) {
+            // 'listType' is the GraphQL type that the list resolves to
+            const listType = fieldType.ofType;
+            if (
+                (listType.toString() === 'Int' ||
+                    listType.toString() === 'String' ||
+                    listType.toString() === 'ID' ||
+                    listType.toString() === 'Boolean' ||
+                    listType.toString() === 'Float') &&
+                typeWeights.scalar === 0 // list won't compound if weight is zero
+            ) {
+                result.fields[field] = {
+                    weight: typeWeights.scalar || DEFAULT_SCALAR_WEIGHT,
+                };
+            } else if (isEnumType(listType) && typeWeights.scalar === 0) {
+                // list won't compound if weight of enum is zero
+                result.fields[field] = {
+                    resolveTo: listType.toString().toLocaleLowerCase(),
+                };
+            } else {
+                fields[field].args.forEach((arg: GraphQLArgument) => {
+                    // If field has an argument matching one of the limiting keywords and resolves to a list
+                    // then the weight of the field should be dependent on both the weight of the resolved type and the limiting argument.
+                    // FIXME: Can nonnull wrap list types?
+                    if (KEYWORDS.includes(arg.name)) {
+                        // Get the type that comprises the list
+                        result.fields[field] = {
+                            resolveTo: listType.toString().toLocaleLowerCase(),
```

Comment:
Does if the list elements are non-null is resolveTo still set correctly? Will double check tests for this case as well. If so, we can address here or with #48

Reply:
Ok. we'll handle it later.